packages/vite/src/node/optimizer/scan.ts

```diff
@@ -171,6 +248,7 @@ export function scanImports(config: ResolvedConfig): {
 async function computeEntries(config: ResolvedConfig) {
   let entries: string[] = []
 
+  // TODO: Should entries be per-environment?
   const explicitEntryPatterns = config.optimizeDeps.entries
```

Comment:
This part of the scanner is now using environment options instead of the global config after https://github.com/vitejs/vite/pull/16471/commits/5e84c51bf2fe438f22c6d6933ded80241f63f2b1. We are still only using the scanner for the client, but if we want to change that this should be closer to support it.